From ee18525521a69742e971a7cca00fb48394df2249 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <67730802+ruslanbay@users.noreply.github.com>
Date: Mon, 9 Mar 2026 15:08:18 +0000
Subject: [PATCH 1/6] patch: arm64: dts: exynos: add initial support for
 Exynos9810 and  Galaxy S9

Introduce comprehensive device tree support for the Exynos 9810 SoC and
the Samsung Galaxy S9 (starlte). This squash includes:

 - Core SoC infrastructure: CMU, sysreg, USI blocks, MCT, RTC, and WDT.
 - Power & Pinctrl: S2DOS05/MAX77705 PMIC integration and pinctrl groups.
 - Connectivity & Storage: UFS, SD card, and USB DRD (with PHY) support.
 - Multimedia & UI: DECON/DSIM display pipeline, S6E3HA8 panel, and
  S6SY761 touchscreen nodes.
 - Platform: Thermal zones, SYSMMU, ramoops, and Wi-Fi/BT board wiring.
---
 ...-add-initial-support-for-Exynos9810-.patch | 869 ++++++++++++++++++
 1 file changed, 869 insertions(+)
 create mode 100644 patches/0001-arm64-dts-exynos-add-initial-support-for-Exynos9810-.patch

diff --git a/patches/0001-arm64-dts-exynos-add-initial-support-for-Exynos9810-.patch b/patches/0001-arm64-dts-exynos-add-initial-support-for-Exynos9810-.patch
new file mode 100644
index 0000000000..e6bf7095d3
--- /dev/null
+++ b/patches/0001-arm64-dts-exynos-add-initial-support-for-Exynos9810-.patch
@@ -0,0 +1,869 @@
+From 77f839d2c3ccc2cd4c141028e075e7b6f6481d94 Mon Sep 17 00:00:00 2001
+From: Ruslan Bay <ruslanbey@proton.me>
+Date: Mon, 9 Mar 2026 14:51:39 +0100
+Subject: [PATCH] arm64: dts: exynos: add initial support for Exynos9810 and
+ Galaxy S9
+
+Introduce comprehensive device tree support for the Exynos 9810 SoC and
+the Samsung Galaxy S9 (starlte). This squash includes:
+
+ - Core SoC infrastructure: CMU, sysreg, USI blocks, MCT, RTC, and WDT.
+ - Power & Pinctrl: S2DOS05/MAX77705 PMIC integration and pinctrl groups.
+ - Connectivity & Storage: UFS, SD card, and USB DRD (with PHY) support.
+ - Multimedia & UI: DECON/DSIM display pipeline, S6E3HA8 panel, and
+  S6SY761 touchscreen nodes.
+ - Platform: Thermal zones, SYSMMU, ramoops, and Wi-Fi/BT board wiring.
+---
+ .../boot/dts/exynos/exynos9810-pinctrl.dtsi   | 106 ++++++
+ .../boot/dts/exynos/exynos9810-starlte.dts    | 343 +++++++++++++++++-
+ arch/arm64/boot/dts/exynos/exynos9810.dtsi    | 279 ++++++++++++++
+ 3 files changed, 726 insertions(+), 2 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/exynos/exynos9810-pinctrl.dtsi b/arch/arm64/boot/dts/exynos/exynos9810-pinctrl.dtsi
+index 88091bf09e4e..8e00426fde01 100644
+--- a/arch/arm64/boot/dts/exynos/exynos9810-pinctrl.dtsi
++++ b/arch/arm64/boot/dts/exynos/exynos9810-pinctrl.dtsi
+@@ -100,6 +100,26 @@ gpq0: gpq0-gpio-bank {
+ 		interrupt-controller;
+ 		#interrupt-cells = <2>;
+ 	};
++
++	pcie_wake: pcie-wake-pins {
++		samsung,pins = "gpa3-2";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_INPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
++
++	wlan_host_wake: wlan-host-wake-pins {
++		samsung,pins = "gpa0-7";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_EINT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_DOWN>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++		samsung,pin-pud-pdn = <EXYNOS_PIN_PULL_DOWN>;
++	};
++
++	bt_hostwake: bt-hostwake-pins {
++		samsung,pins = "gpa2-3";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_INPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
+ };
+ 
+ &pinctrl_aud {
+@@ -356,6 +376,20 @@ gpf0: gpf0-gpio-bank {
+ 		interrupt-controller;
+ 		#interrupt-cells = <2>;
+ 	};
++
++	ufs_rst_n: ufs-rst-n-pins {
++		samsung,pins = "gpf0-1";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_3>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++	};
++
++	ufs_refclk_out: ufs-refclk-out-pins {
++		samsung,pins = "gpf0-0";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_3>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++	};
+ };
+ 
+ &pinctrl_fsys1 {
+@@ -374,6 +408,38 @@ gpf2: gpf2-gpio-bank {
+ 		interrupt-controller;
+ 		#interrupt-cells = <2>;
+ 	};
++
++	pcie0_clkreq: pcie0-clkreq-pins {
++		samsung,pins = "gpf2-0";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++		samsung,pin-pud-pdn = <EXYNOS_PIN_PULL_UP>;
++	};
++
++	pcie0_perst: pcie0-perst-pins {
++		samsung,pins = "gpf2-2";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++	};
++
++	pcie1_clkreq: pcie1-clkreq-pins {
++		samsung,pins = "gpf2-3";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++		samsung,pin-pud-pdn = <EXYNOS_PIN_PULL_UP>;
++	};
++
++	pcie1_perst: pcie1-perst-pins {
++		samsung,pins = "gpf2-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++	};
+ };
+ 
+ &pinctrl_peric0 {
+@@ -432,6 +498,22 @@ gpp3: gpp3-gpio-bank {
+ 		interrupt-controller;
+ 		#interrupt-cells = <2>;
+ 	};
++
++	cfg_wlanen: cfg-wlanen-pins {
++		samsung,pins = "gpg0-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++	};
++
++	bt_btwake: bt-btwake-pins {
++		samsung,pins = "gpg2-0";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_PREV>;
++		samsung,pin-pud-pdn = <EXYNOS_PIN_PULL_NONE>;
++	};
+ };
+ 
+ &pinctrl_peric1 {
+@@ -490,6 +572,30 @@ gpp6: gpp6-gpio-bank {
+ 		interrupt-controller;
+ 		#interrupt-cells = <2>;
+ 	};
++
++	fimc_is_mclk0_fn: fimc-is-mclk0-fn-pins {
++		samsung,pins = "gpc1-4";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
++
++	fimc_is_mclk1_fn: fimc-is-mclk1-fn-pins {
++		samsung,pins = "gpc1-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
++
++	fimc_is_mclk2_fn: fimc-is-mclk2-fn-pins {
++		samsung,pins = "gpc1-6";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
++
++	fimc_is_mclk3_fn: fimc-is-mclk3-fn-pins {
++		samsung,pins = "gpc1-7";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++	};
+ };
+ 
+ &pinctrl_vts {
+diff --git a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
+index fc0ddfee4cd6..1d7fcdb904c6 100644
+--- a/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
++++ b/arch/arm64/boot/dts/exynos/exynos9810-starlte.dts
+@@ -10,12 +10,18 @@
+ #include "exynos9810.dtsi"
+ #include <dt-bindings/gpio/gpio.h>
+ #include <dt-bindings/input/input.h>
++#include <dt-bindings/interrupt-controller/irq.h>
++#include <dt-bindings/usb/pd.h>
+ 
+ / {
+ 	model = "Samsung Galaxy S9 (SM-G960F)";
+ 	compatible = "samsung,starlte", "samsung,exynos9810";
+ 	chassis-type = "handset";
+ 
++	aliases {
++		mmc0 = &mmc;
++	};
++
+ 	chosen {
+ 		#address-cells = <2>;
+ 		#size-cells = <1>;
+@@ -23,7 +29,7 @@ chosen {
+ 
+ 		framebuffer@cc000000 {
+ 			compatible = "simple-framebuffer";
+-			reg = <0x0 0xcc000000 (1440 * 2960 * 4)>;
++			memory-region = <&cont_splash_mem>;
+ 			width = <1440>;
+ 			height = <2960>;
+ 			stride = <(1440 * 4)>;
+@@ -77,17 +83,269 @@ reserved-memory {
+ 		#size-cells = <1>;
+ 		ranges;
+ 
+-		framebuffer@cc000000 {
++		ramoops@92000000 {
++			compatible = "ramoops";
++			reg = <0 0x92000000 0x80000>;
++			record-size = <0x40000>;
++			console-size = <0x40000>;
++		};
++
++		cont_splash_mem: framebuffer@cc000000 {
+ 			reg = <0x0 0xcc000000 (1440 * 2960 * 4)>;
+ 			no-map;
+ 		};
+ 	};
++
++	reg_placeholder: regulator-0 {
++		compatible = "regulator-fixed";
++		regulator-name = "reg-placeholder";
++	};
++
++	ufs_vcc_reg: regulator-1 {
++		compatible = "regulator-fixed";
++		regulator-name = "ufs-vcc";
++		regulator-min-microvolt = <2950000>;
++		regulator-max-microvolt = <2950000>;
++		regulator-boot-on;
++		enable-active-high;
++		gpio = <&gpf0 2 GPIO_ACTIVE_HIGH>;
++	};
++
++	battery: battery {
++		compatible = "simple-battery";
++		constant-charge-current-max-microamp = <2150000>;
++		charge-full-design-microamp-hours = <3000000>;
++		voltage-min-design-microvolt = <3400000>;
++		voltage-max-design-microvolt = <4350000>;
++	};
+ };
+ 
+ &oscclk {
+ 	clock-frequency = <26000000>;
+ };
+ 
++&hsi2c_43 {
++	#address-cells = <1>;
++	#size-cells = <0>;
++	status = "okay";
++
++	s2dos05_pmic: pmic@60 {
++		compatible = "samsung,s2dos05";
++		reg = <0x60>;
++
++		regulators {
++			s2dos05_ldo1: ldo1 {
++				regulator-name = "s2dos05-ldo1";
++				regulator-min-microvolt = <1500000>;
++				regulator-max-microvolt = <2000000>;
++			};
++
++			s2dos05_ldo2: ldo2 {
++				regulator-name = "s2dos05-ldo2";
++				regulator-min-microvolt = <1800000>;
++				regulator-max-microvolt = <1800000>;
++				regulator-boot-on;
++			};
++
++			s2dos05_ldo3: ldo3 {
++				regulator-name = "s2dos05-ldo3";
++				regulator-min-microvolt = <3000000>;
++				regulator-max-microvolt = <3000000>;
++				regulator-boot-on;
++			};
++
++			s2dos05_ldo4: ldo4 {
++				regulator-name = "s2dos05-ldo4";
++				regulator-min-microvolt = <2700000>;
++				regulator-max-microvolt = <3775000>;
++			};
++
++			s2dos05_buck: buck {
++				regulator-name = "s2dos05-buck";
++				regulator-min-microvolt = <850000>;
++				regulator-max-microvolt = <2100000>;
++			};
++		};
++	};
++};
++
++&hsi2c_10 {
++	#address-cells = <1>;
++	#size-cells = <0>;
++	status = "okay";
++
++	touchscreen@48 {
++		compatible = "samsung,s6sy761";
++		reg = <0x48>;
++		interrupt-parent = <&gpa1>;
++		interrupts = <0 IRQ_TYPE_LEVEL_LOW>;
++		avdd-supply = <&s2dos05_ldo3>;
++		vdd-supply = <&s2dos05_ldo2>;
++		pinctrl-0 = <&ts_int>;
++		pinctrl-names = "default";
++	};
++};
++
++&hsi2c_11 {
++	#address-cells = <1>;
++	#size-cells = <0>;
++	status = "okay";
++
++	nfc@27 {
++		compatible = "samsung,s3fwrn5-i2c";
++		reg = <0x27>;
++		pinctrl-names = "default";
++		pinctrl-0 = <&nfc_irq &nfc_en &nfc_wake>;
++		interrupt-parent = <&gpa1>;
++		interrupts = <2 IRQ_TYPE_EDGE_RISING>;
++		en-gpios = <&gpc1 3 GPIO_ACTIVE_HIGH>;
++		wake-gpios = <&gpc1 1 GPIO_ACTIVE_HIGH>;
++		status = "disabled";
++	};
++
++	imu@6a {
++		compatible = "st,lsm6dsl";
++		reg = <0x6a>;
++		interrupt-parent = <&gpa1>;
++		interrupts = <1 IRQ_TYPE_EDGE_RISING>;
++		vdd-supply = <&reg_placeholder>;
++		vddio-supply = <&reg_placeholder>;
++		status = "disabled";
++	};
++};
++
++&hsi2c_44 {
++	#address-cells = <1>;
++	#size-cells = <0>;
++	status = "okay";
++
++	max77705: pmic@66 {
++		compatible = "maxim,max77705";
++		reg = <0x66>;
++		interrupt-controller;
++		#interrupt-cells = <1>;
++		interrupt-parent = <&gpa1>;
++		interrupts = <2 IRQ_TYPE_LEVEL_LOW>;
++	};
++
++	max77705_charger: charger@69 {
++		compatible = "maxim,max77705-charger";
++		reg = <0x69>;
++		monitored-battery = <&battery>;
++		interrupt-parent = <&max77705>;
++		interrupts = <0>;
++	};
++
++	fuel-gauge@36 {
++		compatible = "maxim,max77705-battery";
++		reg = <0x36>;
++		power-supplies = <&max77705_charger>;
++		interrupt-parent = <&max77705>;
++		interrupts = <2>;
++	};
++};
++
++&ufs {
++	status = "okay";
++	vcc-supply = <&ufs_vcc_reg>;
++	pinctrl-0 = <&ufs_rst_n &ufs_refclk_out>;
++};
++
++&ufs_phy {
++	status = "okay";
++};
++
++&mmc {
++	pinctrl-0 = <&sd2_clk &sd2_cmd &sd2_bus1 &sd2_bus4 &sd2_cd>;
++	pinctrl-names = "default";
++	bus-width = <4>;
++	cd-gpios = <&gpa1 5 GPIO_ACTIVE_LOW>;
++	card-detect-delay = <200>;
++	clock-frequency = <800000000>;
++	disable-wp;
++	sd-uhs-sdr50;
++	sd-uhs-sdr104;
++	status = "okay";
++
++	wifi@1 {
++		compatible = "brcm,bcm4359-fmac", "brcm,bcm4329-fmac";
++		reg = <1>;
++		interrupt-parent = <&gpa0>;
++		interrupts = <7 IRQ_TYPE_LEVEL_LOW>;
++		interrupt-names = "host-wake";
++		reset-gpios = <&gpg0 5 GPIO_ACTIVE_LOW>;
++	};
++};
++
++&usbdrd {
++	status = "okay";
++};
++
++&usbdrd_dwc3 {
++	status = "okay";
++	dr_mode = "otg";
++	#address-cells = <1>;
++	#size-cells = <0>;
++
++	connector {
++		compatible = "usb-c-connector";
++		label = "USB-C";
++		data-role = "dual";
++		power-role = "dual";
++		try-power-role = "sink";
++		source-pdos = <PDO_FIXED(5000, 900,
++					 (PDO_FIXED_SUSPEND | PDO_FIXED_USB_COMM |
++					  PDO_FIXED_DATA_SWAP | PDO_FIXED_DUAL_ROLE))>;
++		sink-pdos = <PDO_FIXED(5000, 3000,
++				       (PDO_FIXED_DATA_SWAP | PDO_FIXED_USB_COMM |
++					PDO_FIXED_HIGHER_CAP | PDO_FIXED_DUAL_ROLE))>;
++	};
++};
++
++&usbdrd_phy {
++	status = "okay";
++};
++
++&dsim0 {
++	status = "okay";
++
++	panel@0 {
++		compatible = "samsung,s6e3ha8";
++		reg = <0>;
++		vci-supply = <&s2dos05_ldo4>;
++		vddr-supply = <&s2dos05_buck>;
++		vdd3-supply = <&s2dos05_ldo1>;
++		reset-gpios = <&gpg0 4 GPIO_ACTIVE_HIGH>;
++
++		port {
++			panel_in: endpoint {
++				remote-endpoint = <&dsim0_out>;
++			};
++		};
++	};
++};
++
++&pinctrl_peric0 {
++	panel_reset: panel-reset-pins {
++		samsung,pins = "gpg0-4";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV3>;
++	};
++};
++
++/ {
++	bluetooth {
++		compatible = "brcm,bcm4329-bt";
++		shutdown-gpios = <&gpg2 0 GPIO_ACTIVE_HIGH>;
++		device-wakeup-gpios = <&gpg2 0 GPIO_ACTIVE_HIGH>;
++		host-wakeup-gpios = <&gpa2 3 GPIO_ACTIVE_HIGH>;
++		pinctrl-0 = <&bt_hostwake &bt_btwake>;
++		pinctrl-names = "default";
++		status = "disabled";
++	};
++};
++
+ &pinctrl_alive {
+ 	key_power: key-power-pins {
+ 		samsung,pins = "gpa2-4";
+@@ -116,4 +374,85 @@ key_wink: key-wink-pins {
+ 		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
+ 		samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+ 	};
++
++	sd2_cd: sd2-cd-pins {
++		samsung,pins = "gpa1-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_EINT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
++	};
++
++	ts_int: ts-int-pins {
++		samsung,pins = "gpa1-0";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_EINT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++
++	nfc_irq: nfc-irq-pins {
++		samsung,pins = "gpa1-2";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_EINT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++};
++
++&pinctrl_peric1 {
++	nfc_en: nfc-en-pins {
++		samsung,pins = "gpc1-3";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++
++	nfc_wake: nfc-wake-pins {
++		samsung,pins = "gpc1-1";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++
++	cam_mclk0: cam-mclk0-pins {
++		samsung,pins = "gpc1-4";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++
++	cam_mclk1: cam-mclk1-pins {
++		samsung,pins = "gpc1-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV1>;
++	};
++};
++
++&pinctrl_fsys1 {
++	sd2_clk: sd2-clk-pins {
++		samsung,pins = "gpf1-0";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
++	};
++
++	sd2_cmd: sd2-cmd-pins {
++		samsung,pins = "gpf1-1";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
++	};
++
++	sd2_bus1: sd2-bus-width1-pins {
++		samsung,pins = "gpf1-2", "gpf1-3", "gpf1-4", "gpf1-5";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
++	};
++
++	sd2_bus4: sd2-bus-width4-pins {
++		samsung,pins = "gpf1-6", "gpf1-7", "gpf2-0", "gpf2-1";
++		samsung,pin-function = <EXYNOS_PIN_FUNC_2>;
++		samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
++		samsung,pin-drv = <EXYNOS7_PIN_DRV_LV4>;
++	};
+ };
+diff --git a/arch/arm64/boot/dts/exynos/exynos9810.dtsi b/arch/arm64/boot/dts/exynos/exynos9810.dtsi
+index 01eba31f7ca3..2822b1867374 100644
+--- a/arch/arm64/boot/dts/exynos/exynos9810.dtsi
++++ b/arch/arm64/boot/dts/exynos/exynos9810.dtsi
+@@ -7,6 +7,7 @@
+  */
+ 
+ #include <dt-bindings/interrupt-controller/arm-gic.h>
++#include <dt-bindings/soc/samsung,exynos-usi.h>
+ 
+ / {
+ 	compatible = "samsung,exynos9810";
+@@ -171,6 +172,45 @@ chipid@10000000 {
+ 			reg = <0x10000000 0x100>;
+ 		};
+ 
++		cmu_peric0: clock-controller@10400000 {
++			compatible = "samsung,exynos9810-cmu-peric0";
++			reg = <0x10400000 0x8000>;
++			#clock-cells = <1>;
++			clocks = <&oscclk>;
++			clock-names = "oscclk";
++		};
++
++		cmu_peric1: clock-controller@10800000 {
++			compatible = "samsung,exynos9810-cmu-peric1";
++			reg = <0x10800000 0x8000>;
++			#clock-cells = <1>;
++			clocks = <&oscclk>;
++			clock-names = "oscclk";
++		};
++
++		cmu_fsys1: clock-controller@11400000 {
++			compatible = "samsung,exynos9810-cmu-fsys1";
++			reg = <0x11400000 0x8000>;
++			#clock-cells = <1>;
++			clocks = <&oscclk>;
++			clock-names = "oscclk";
++		};
++
++		syscon_peric0: syscon@10420000 {
++			compatible = "samsung,exynos9810-peric0-sysreg", "syscon";
++			reg = <0x10420000 0x2000>;
++		};
++
++		syscon_peric1: syscon@10820000 {
++			compatible = "samsung,exynos9810-peric1-sysreg", "syscon";
++			reg = <0x10820000 0x2000>;
++		};
++
++		syscon_fsys1: syscon@11420000 {
++			compatible = "samsung,exynos9810-fsys1-sysreg", "syscon";
++			reg = <0x11420000 0x2000>;
++		};
++
+ 		gic: interrupt-controller@10101000 {
+ 			compatible = "arm,gic-400";
+ 			reg = <0x10101000 0x1000>,
+@@ -252,6 +292,224 @@ pinctrl_aud: pinctrl@17c60000 {
+ 			compatible = "samsung,exynos9810-pinctrl";
+ 			reg = <0x17c60000 0x1000>;
+ 		};
++
++		usi_hsi2c_10: usi@104b00c0 {
++			compatible = "samsung,exynos850-usi";
++			reg = <0x104b00c0 0x20>;
++			samsung,mode = <USI_MODE_I2C>;
++			#address-cells = <1>;
++			#size-cells = <1>;
++			ranges;
++			status = "disabled";
++
++			hsi2c_10: i2c@104b0000 {
++				compatible = "samsung,exynos850-hsi2c",
++					     "samsung,exynosautov9-hsi2c";
++				reg = <0x104b0000 0xc0>;
++				interrupts = <GIC_SPI 404 IRQ_TYPE_LEVEL_HIGH>;
++				#address-cells = <1>;
++				#size-cells = <0>;
++				status = "disabled";
++			};
++		};
++
++		usi_hsi2c_11: usi@108e00c0 {
++			compatible = "samsung,exynos850-usi";
++			reg = <0x108e00c0 0x20>;
++			samsung,mode = <USI_MODE_I2C>;
++			#address-cells = <1>;
++			#size-cells = <1>;
++			ranges;
++			status = "disabled";
++
++			hsi2c_11: i2c@108e0000 {
++				compatible = "samsung,exynos850-hsi2c",
++					     "samsung,exynosautov9-hsi2c";
++				reg = <0x108e0000 0xc0>;
++				interrupts = <GIC_SPI 432 IRQ_TYPE_LEVEL_HIGH>;
++				#address-cells = <1>;
++				#size-cells = <0>;
++				status = "disabled";
++			};
++		};
++
++		usi_hsi2c_43: usi@143500c0 {
++			compatible = "samsung,exynos850-usi";
++			reg = <0x143500c0 0x20>;
++			samsung,mode = <USI_MODE_I2C>;
++			#address-cells = <1>;
++			#size-cells = <1>;
++			ranges;
++			status = "disabled";
++
++			hsi2c_43: i2c@14350000 {
++				compatible = "samsung,exynos850-hsi2c",
++					     "samsung,exynosautov9-hsi2c";
++				reg = <0x14350000 0xc0>;
++				interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>;
++				#address-cells = <1>;
++				#size-cells = <0>;
++				status = "disabled";
++			};
++		};
++
++		usi_hsi2c_44: usi@143600c0 {
++			compatible = "samsung,exynos850-usi";
++			reg = <0x143600c0 0x20>;
++			samsung,mode = <USI_MODE_I2C>;
++			#address-cells = <1>;
++			#size-cells = <1>;
++			ranges;
++			status = "disabled";
++
++			hsi2c_44: i2c@14360000 {
++				compatible = "samsung,exynos850-hsi2c",
++					     "samsung,exynosautov9-hsi2c";
++				reg = <0x14360000 0xc0>;
++				interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
++				#address-cells = <1>;
++				#size-cells = <0>;
++				status = "disabled";
++			};
++		};
++
++		mct: timer@10040000 {
++			compatible = "samsung,exynos8895-mct",
++				     "samsung,exynos4210-mct";
++			reg = <0x10040000 0x800>;
++			clocks = <&oscclk>, <&oscclk>;
++			clock-names = "fin_pll", "mct";
++			interrupts = <GIC_SPI 455 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 456 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 457 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 458 IRQ_TYPE_LEVEL_HIGH>;
++			status = "disabled";
++		};
++
++		rtc: rtc@10590000 {
++			compatible = "samsung,exynos7-rtc", "samsung,s3c6410-rtc";
++			reg = <0x10590000 0x100>;
++			interrupts = <GIC_SPI 355 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 356 IRQ_TYPE_LEVEL_HIGH>;
++			clocks = <&oscclk>;
++			clock-names = "rtc";
++			status = "disabled";
++		};
++
++		watchdog_cl0: watchdog@10050000 {
++			compatible = "samsung,exynos7-wdt";
++			reg = <0x10050000 0x100>;
++			interrupts = <GIC_SPI 460 IRQ_TYPE_LEVEL_HIGH>;
++			clocks = <&oscclk>;
++			clock-names = "watchdog";
++			samsung,syscon-phandle = <&pmu_system_controller>;
++			status = "disabled";
++		};
++
++		watchdog_cl1: watchdog@10060000 {
++			compatible = "samsung,exynos7-wdt";
++			reg = <0x10060000 0x100>;
++			interrupts = <GIC_SPI 461 IRQ_TYPE_LEVEL_HIGH>;
++			clocks = <&oscclk>;
++			clock-names = "watchdog";
++			samsung,syscon-phandle = <&pmu_system_controller>;
++			status = "disabled";
++		};
++
++		ufs: ufs@11120000 {
++			compatible = "samsung,exynos-ufs";
++			reg = <0x11120000 0x200>,
++			      <0x11121100 0x200>,
++			      <0x11110000 0x8000>,
++			      <0x11130000 0x100>;
++			interrupts = <GIC_SPI 242 IRQ_TYPE_LEVEL_HIGH>;
++			pinctrl-names = "default";
++			pinctrl-0 = <&ufs_rst_n &ufs_refclk_out>;
++			status = "disabled";
++		};
++
++		ufs_phy: phy@11124000 {
++			compatible = "samsung,exynos7-ufs-phy";
++			reg = <0x11124000 0x800>;
++			#phy-cells = <0>;
++			status = "disabled";
++		};
++
++		mmc: mmc@11500000 {
++			compatible = "samsung,exynos-dw-mshc";
++			reg = <0x11500000 0x2000>;
++			interrupts = <GIC_SPI 249 IRQ_TYPE_LEVEL_HIGH>;
++			#address-cells = <1>;
++			#size-cells = <0>;
++			status = "disabled";
++		};
++
++		usbdrd: usb@10c00000 {
++			compatible = "samsung,exynos850-dwusb3";
++			ranges = <0x0 0x10c00000 0x10000>;
++			clocks = <&oscclk>, <&oscclk>;
++			clock-names = "bus_early", "ref";
++			#address-cells = <1>;
++			#size-cells = <1>;
++			status = "disabled";
++
++			usbdrd_dwc3: usb@0 {
++				compatible = "snps,dwc3";
++				reg = <0x0 0x10000>;
++				interrupts = <GIC_SPI 244 IRQ_TYPE_LEVEL_HIGH>;
++				phys = <&usbdrd_phy 0>;
++				phy-names = "usb2-phy";
++			};
++		};
++
++		usbdrd_phy: phy@11100000 {
++			compatible = "samsung,exynos850-usbdrd-phy";
++			reg = <0x11100000 0x100>;
++			clocks = <&oscclk>, <&oscclk>;
++			clock-names = "phy", "ref";
++			samsung,pmu-syscon = <&pmu_system_controller>;
++			#phy-cells = <1>;
++			status = "disabled";
++		};
++
++		decon_f: display-controller@16030000 {
++			compatible = "samsung,exynos9-decon";
++			reg = <0x16030000 0x10000>;
++			interrupts = <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>;
++			status = "disabled";
++		};
++
++		dsim0: dsi@16080000 {
++			compatible = "samsung,exynos9-dsim";
++			reg = <0x16080000 0x10000>;
++			interrupts = <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH>;
++			#address-cells = <1>;
++			#size-cells = <0>;
++			status = "disabled";
++
++			port {
++				dsim0_out: endpoint {
++					remote-endpoint = <&panel_in>;
++				};
++			};
++		};
++
++		sysmmu_pcie: iommu@11420000 {
++			compatible = "samsung,pcie-sysmmu";
++			reg = <0x11420000 0x9000>;
++			interrupts = <GIC_SPI 264 IRQ_TYPE_LEVEL_HIGH>;
++			#iommu-cells = <0>;
++			status = "disabled";
++		};
++
++		sysmmu_dpu: iommu@160a0000 {
++			compatible = "samsung,exynos-sysmmu";
++			reg = <0x160a0000 0x9000>;
++			interrupts = <GIC_SPI 211 IRQ_TYPE_LEVEL_HIGH>,
++				     <GIC_SPI 210 IRQ_TYPE_LEVEL_HIGH>;
++			#iommu-cells = <0>;
++			status = "disabled";
++		};
+ 	};
+ 
+ 	timer {
+@@ -267,6 +525,27 @@ timer {
+ 		 */
+ 		clock-frequency = <26000000>;
+ 	};
++
++	thermal-zones {
++		cpu-thermal {
++			polling-delay-passive = <250>;
++			polling-delay = <1000>;
++
++			trips {
++				cpu_alert0: cpu-alert0 {
++					temperature = <75000>;
++					hysteresis = <2000>;
++					type = "passive";
++				};
++
++				cpu_crit0: cpu-crit0 {
++					temperature = <115000>;
++					hysteresis = <0>;
++					type = "critical";
++				};
++			};
++		};
++	};
+ };
+ 
+ #include "exynos9810-pinctrl.dtsi"
+-- 
+2.52.0
+
-- 
2.43.0

